src/e2e/utils/helpers.ts

```diff
@@ -76,23 +76,31 @@ export const getVerificationCode = async (
   testEmail: string,
   page: Page,
   attempts = 10,
-): Promise<string> => {
+): Promise<string | undefined> => {
   if (attempts === 0) {
     throw new Error("Unable to retrieve restmail data");
   }
 
-  const context = await request.newContext();
-  const res = await context.get(`http://restmail.net/mail/${testEmail}`, {
-    failOnStatusCode: false,
-  });
-  const resJson = await res.json();
-  if (resJson.length) {
+  try {
```

Comment:
Handle unexpected errors and increase waiting time for retrieving the verification code from restmail.